### PR TITLE
use the plugin, sort out a few sendable issues in the PID interfaces

### PR DESCRIPTION
**Pull request by laviRZ**

### CODE REVIEW

File: src/main/java/frc/robot/utilities/pid/PIDFMotor.java [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+import edu.wpi.first.util.sendable.SendableBuilder;
 
 /**
  * An interface used for motors that have a configurable PID
  */
 public interface PIDFMotor extends PIDFConfigurable {
     void setCoefs(PIDFCoefs pidfCoefs);
 
-    void tunePIDF(ControlMode controlMode);
+    void setTuning(ControlMode controlMode);
```
Comment (Mboesn) [inline src/main/java/frc/robot/utilities/pid/PIDFMotor.java:12]: is setTuning really a good name?

File: src/main/java/frc/robot/utilities/pid/PIDFTalonFX.java [hunk 2]
```diff
@@ -51,4 +52,8 @@ public void setSetpoint(double setpoint) {
     public boolean isTuning() {
         return isTuning;
     }
+
+    public void initSendable(SendableBuilder builder) {
```
Comment (Mboesn) [inline src/main/java/frc/robot/utilities/pid/PIDFTalonFX.java:56]: @ overide

File: src/main/java/frc/robot/utilities/pid/PIDFTalonSRX.java [hunk 3]
```diff
@@ -51,4 +52,8 @@ public void setSetpoint(double setpoint) {
     public boolean isTuning() {
         return isTuning;
     }
+
+    public void initSendable(SendableBuilder builder) {
```
Comment (Mboesn) [inline src/main/java/frc/robot/utilities/pid/PIDFTalonSRX.java:56]: @ override

File: src/main/java/frc/robot/utilities/pid/TrigonPIDController.java [hunk 4]
```diff
@@ -57,4 +58,8 @@ public void setDeltaTolerance(double deltaTolerance) {
         pidCoefs.setDeltaTolerance(deltaTolerance);
         super.setTolerance(pidCoefs.getTolerance(), deltaTolerance);
     }
+
+    public void initSendable(SendableBuilder builder) {
```
Comment (Mboesn) [inline src/main/java/frc/robot/utilities/pid/TrigonPIDController.java:62]: @ override

File: src/main/java/frc/robot/utilities/pid/TrigonPIDFController.java [hunk 5]
```diff
@@ -42,4 +44,9 @@ public void setKS(double s) {
     public PIDFCoefs getCoefs() {
         return pidfCoefs;
     }
+
+    @Override
+    public void initSendable(SendableBuilder builder) {
```
Comment (Mboesn) [inline src/main/java/frc/robot/utilities/pid/TrigonPIDFController.java:49]: @ override

File: src/main/java/frc/robot/RobotContainer.java [hunk 6]
```diff
@@ -27,6 +30,9 @@ public RobotContainer() {
         initializeSubsystems();
         initializeCommands();
         bindCommands();
+        SmartDashboard.putData(new TrigonPIDController(new PIDCoefs()));
+        SmartDashboard.putData(new TrigonPIDFController(new PIDFCoefs()));
+        SmartDashboard.putData(new PIDFTalonFX(0, new MotorConfig()));
```
Comment (Mboesn) [inline src/main/java/frc/robot/RobotContainer.java:35]: remove, if you want you can send the modules

File: src/main/java/frc/robot/utilities/pid/PIDFMotor.java (hunk 1 above)
Comment (lycaon7) [reply src/main/java/frc/robot/utilities/pid/PIDFMotor.java:12]: no it isnt

**Pull request closed without merging**